File: python/bank.py
# OOP 
class BankAccount:

    def __init__(self, account_holder, account_num, balance=0) :
        self.account_holder = account_holder
        self.account_num = account_num
        self.balance = balance

    def deposit(self):
        deposit_amt = int(input("Enter an amount to deposit: "))
        if deposit_amt > 0:
            question = input(f"Are you sure you want to deposit {deposit_amt} into your account? (Yes/No) \n")
            if question == "Yes":
                self.balance += deposit_amt
                print(f"You have deposited {deposit_amt} into your account, your current balance is {self.balance}")
            else:
                print("Transaction failed")
        else:
            print("Invalid input")

    def withdraw(self):
        withdraw_amt = int(input("Enter amount to withdraw: "))
        question = input (f"Are you sure you want to withdraw {withdraw_amt} from your account? (Yes/No) \n")
        rem_balance = self.balance
        
        if question == "Yes":
            if withdraw_amt > 0:
                if withdraw_amt <= self.balance:
                    rem_balance = self.balance - withdraw_amt
                    print(f"You have withdrawn {withdraw_amt} from your account, your balance is {rem_balance}")
                else: 
                    print("Insufficient funds")
            else:
                print("Cannot withdraw amount")
        else:
            print("Transaction failed!")

    def display_details(self):
        print(f"Account number : {self.account_num} \n Account name: {self.account_holder} \n Balance: {self.balance} ")

    def check_balance(self):
        print (f"Your current balance: {self.balance}") 

File: java/bank.java
import java.util.Scanner;

public class bank {
    String account_holder;
    int account_number;
    int account_balance=0;

    Scanner input = new Scanner(System.in);

        public void deposit() {
            System.out.print("Enter an amount to deposit: ");
            int deposit_amount = input.nextInt();
            input.nextLine();

            if (deposit_amount > 0){
                System.out.print("Are you sure you want to deposit this amount into your account? (Yes/No): ");
                String question = input.nextLine();
                if (question.equals("Yes")){
                    this.account_balance+=deposit_amount;
                    System.out.printf("You have deposited %d into your account", deposit_amount);
                }
                else{
                    System.out.println("Transaction failed");
                }
            }
            else{
                System.out.println("Invalid Input");
            }

            input.close();

        }

        public void withdraw() {
            System.out.print("Enter an amount to withdraw: ");
            int amount = input.nextInt();
            input.nextLine();

            if (amount > 0){
                System.out.print("Are you sure you want to withdraw this amount (Yes/No): ");
                String question = input.nextLine();
                if (question.equals("Yes")){
                    if (amount <= this.account_balance){
                        this.account_balance -= amount;
                        System.out.printf("You have withdrawn %d from your account", amount);
                    }
                    else{
                        System.out.println("Insufficient funds");
                    }
                }
                else{
                    System.out.println("Transaction failed");
                }
            }
            else{
                System.out.print("Cannot withdraw amount");
            }

            input.close();
        }

        public void display_details(){
            System.out.printf(" Account number: %d \n Account name: %s \n Account Balance: %d ", account_number, account_holder, account_balance);
        }

        public void check_balance(){
            System.out.printf(" Your current balance: %d", account_balance);
        }
}
